File: 2026/Company/Company.Web/Modules/ServerTypes/Sales/DocumentTypeRow.ts
import { fieldsProxy, getLookup, getLookupAsync } from "@serenity-is/corelib";

export interface DocumentTypeRow {
    DocumentTypeId?: number;
    DocumentName?: string;
    IsActive?: number;
    TenantId?: number;
    InsertUserId?: number;
    InsertDate?: string;
    UpdateUserId?: number;
    UpdateDate?: string;
}

export abstract class DocumentTypeRow {
    static readonly idProperty = 'DocumentTypeId';
    static readonly isActiveProperty = 'IsActive';
    static readonly nameProperty = 'DocumentName';
    static readonly localTextPrefix = 'Sales.DocumentType';
    static readonly lookupKey = 'Sales.DokumenyYype';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<DocumentTypeRow>('Sales.DokumenyYype') }
    static async getLookupAsync() { return getLookupAsync<DocumentTypeRow>('Sales.DokumenyYype') }

    static readonly deletePermission = 'Company:Currency:Delete';
    static readonly insertPermission = 'Company:Currency:Modify';
    static readonly readPermission = 'Company:Currency:View';
    static readonly updatePermission = 'Company:Currency:Modify';

    static readonly Fields = fieldsProxy<DocumentTypeRow>();
}

File: 2026/Company/Company.Web/Modules/Sales/DocumentType/DocumentTypeRow.cs
using Company.Administration.Entities;
namespace Company.Sales;

[ConnectionKey("Sales"), Module("Sales"), TableName("[sales].[DocumentType]")]
[DisplayName("Document Type"), InstanceName("Document Type")]
[ReadPermission(SalesPermissionKeys.Currency.View)]
[ModifyPermission(SalesPermissionKeys.Currency.Modify)]
[DeletePermission(SalesPermissionKeys.Currency.Delete)]
[ServiceLookupPermission("SalesPermission:General")]
[LookupScript("Sales.DokumenyYype", LookupType = typeof(MultiTenantRowLookupScript<>))]

public sealed class DocumentTypeRow : LoggingRow<DocumentTypeRow.RowFields>, IIdRow, INameRow, IIsActiveRow, IMultiTenantRow
{
    [DisplayName("Document Type Id"), Column("DocumentTypeID"), Identity, IdProperty]
    public int? DocumentTypeId { get => fields.DocumentTypeId[this]; set => fields.DocumentTypeId[this] = value; }

    [DisplayName("Document Name"), Size(50), NotNull, QuickSearch, NameProperty]
    public string DocumentName { get => fields.DocumentName[this]; set => fields.DocumentName[this] = value; }

    #region Tenant & Activ

    [Insertable(false), Updatable(false)]
    public Int32? TenantId { get => fields.TenantId[this]; set => fields.TenantId[this] = value; }

    public Int32Field TenantIdField { get => fields.TenantId; }

    [NotNull, Insertable(false), Updatable(true)]
    public Int16? IsActive { get => fields.IsActive[this]; set => fields.IsActive[this] = value; }

    Int16Field IIsActiveRow.IsActiveField { get => fields.IsActive; }

    #endregion Tenant & Activ

    public class RowFields : LoggingRowFields
    {
        public Int32Field DocumentTypeId;
        public StringField DocumentName;

        public Int16Field IsActive;
        public Int32Field TenantId;
    }
}